File: LeetCode/LeetCode/Problems/1/Problem169.cs
namespace LeetCode.Problems
{
    public class Problem169
    {
        public int MajorityElement(int[] nums)
        {
            var candidate = nums[0];
            var count = 1;

            for (var i = 1; i < nums.Length; i++)
            {
                if (nums[i] == candidate)
                    count++;
                else if (count == 1)
                    candidate = nums[i];
                else
                    count--;
            }

            return candidate;
        }
    }
}

File: leetcode_rust/src/problems/problem_169.rs
pub fn majority_element(nums: Vec<i32>) -> i32 {
    let mut major = i32::MAX;
    let mut count = 1;

    for num in nums {
        if num == major {
            count += 1;
        } else {
            count -= 1;
            if count == 0 {
                major = num;
                count = 1;
            }
        }
    }

    major
}
